### PR TITLE
Some missing stubs for cryptography.x509

### PR DESCRIPTION
**Pull request by jolaf**

### CODE REVIEW

**Comment by jolaf** [comment]
Thanks `mypy_primer`, you've made me add even more stubs. :)

File: third_party/2and3/cryptography/x509/__init__.pyi [hunk 1]
```diff
@@ -286,6 +286,14 @@ class Extensions(object):
...
     def get_extension_for_class(self, extclass: Type[_T]) -> Extension[_T]: ...
 
+class DuplicateExtension(Exception):
+    oid: ObjectIdentifier
+    def __init__(self, msg: str, oid: ObjectIdentifier) -> None: ...
+
+class ExtensionNotFound(Exception):
+    oid: ObjectIdentifier
+    def __init__(self, msg: str, oid: ObjectIdentifier) -> None: ...
+
```
Comment (srittau) [inline third_party/2and3/cryptography/x509/__init__.pyi:296]: Not a full review, but in the implementation, these also live in `extensions.py` and are only re-exported from `__init__.py`.

Comment (jolaf) [reply third_party/2and3/cryptography/x509/__init__.pyi:296]: That's true, but implementation is very weird in this aspects, in contains most things in two places.

So I've checked the documentation and placed things where documentation places them.

**Review by hauntsaninja — COMMENTED** [review]
Thanks for doing this!

I ran this diff through stubtest, and here are the issues it finds. Most look worth fixing:

<details>

```
(typeshed) ~/dev/typeshed cryptography λ stubtest cryptography --custom-typeshed-dir ~/dev/typeshed --whitelist crypbase.txt --concise
cryptography.x509.extensions.AccessDescription.__init__ is inconsistent, stub does not have argument "access_method"
cryptography.x509.extensions.AccessDescription.__init__ is inconsistent, stub does not have argument "access_location"
cryptography.x509.extensions.AuthorityInformationAccess.__init__ is inconsistent, stub does not have argument "descriptions"
cryptography.x509.extensions.AuthorityInformationAccess.oid is not present in stub
cryptography.x509.extensions.CRLDistributionPoints.__init__ is inconsistent, stub does not have argument "distribution_points"
cryptography.x509.extensions.CRLDistributionPoints.oid is not present in stub
cryptography.x509.extensions.DistributionPoint.__init__ is inconsistent, stub does not have argument "full_name"
cryptography.x509.extensions.DistributionPoint.__init__ is inconsistent, stub does not have argument "relative_name"
cryptography.x509.extensions.DistributionPoint.__init__ is inconsistent, stub does not have argument "reasons"
cryptography.x509.extensions.DistributionPoint.__init__ is inconsistent, stub does not have argument "crl_issuer"
cryptography.x509.extensions.DistributionPoint.crl_issuer is not present in stub
cryptography.x509.extensions.DistributionPoint.reasons is not present in stub
cryptography.x509.extensions.DistributionPoint.relative_name is not present in stub
cryptography.x509.extensions.Extension.__init__ is inconsistent, stub does not have argument "oid"
cryptography.x509.extensions.Extension.__init__ is inconsistent, stub does not have argument "critical"
cryptography.x509.extensions.Extension.__init__ is inconsistent, stub does not have argument "value"
cryptography.x509.extensions.Extension.critical is not present in stub
cryptography.x509.extensions.Extension.oid is not present in stub
cryptography.x509.extensions.GeneralNames.__init__ is inconsistent, stub does not have argument "general_names"
cryptography.x509.extensions.GeneralNames.get_values_for_type is not present in stub
cryptography.x509.oid.ExtensionOID.SIGNED_CERTIFICATE_TIMESTAMPS is not present in stub
cryptography.x509.oid.NameOID.INN is not present in stub
cryptography.x509.oid.NameOID.OGRN is not present in stub
cryptography.x509.oid.NameOID.SNILS is not present in stub
cryptography.x509.oid.NameOID.UNSTRUCTURED_NAME is not present in stub
cryptography.x509.oid.SignatureAlgorithmOID.GOSTR3410_2012_WITH_3411_2012_256 is not present in stub
cryptography.x509.oid.SignatureAlgorithmOID.GOSTR3410_2012_WITH_3411_2012_512 is not present in stub
cryptography.x509.oid.SignatureAlgorithmOID.GOSTR3411_94_WITH_3410_2001 is not present in stub
note: unused whitelist entry cryptography.x509.DuplicateExtension
note: unused whitelist entry cryptography.x509.ExtensionNotFound
```
</details>

**Comment by jolaf** [comment]
> I ran this diff through stubtest, and here are the issues it finds.

Thanks for running the test. Unfortunately, I'm out of resource to fix this any further.

My suggestion is merging this PR "as is" and leave further fixing for the future.

**Review by srittau — APPROVED** [review]

**Comment by srittau** [comment]
Sorry for the late reply. A cursory glance shows no red flags and further improvements can always be made. I fixed the merge conflict.

**Comment by jolaf** [comment]
Thanks!

**Pull request merged**